frontend/app/src/pages/lobbySelectionPage/LobbySelectionPage.test.tsx

```diff
@@ -0,0 +1,491 @@
+import { Button } from "@/components/ui/button";
+import { BASE_POLLING_INTERVAL_MS } from "@/consts";
+import { Lobby } from "@/domain/Lobby";
+import { getLobbiesV1 } from "@/services/rest/RestService";
+import { aLobby, mockLobbies } from "@/TestFactories";
+import {
+	act,
+	fireEvent,
+	render,
+	screen,
+	waitFor,
+} from "@testing-library/react";
+import { createMemoryRouter, Navigate, RouterProvider } from "react-router-dom";
+import { beforeEach, describe, expect, it, Mock, vi } from "vitest";
+import LobbySelectionPage from "./LobbySelectionPage";
+import { useDispatchLobby } from "@/services/lobbyContext/LobbyContextHelpers";
+
+vi.mock("@/services/rest/RestService", () => ({
+	getLobbiesV1: vi.fn(),
+}));
+
+vi.mock("@/services/lobbyContext/LobbyContextHelpers", () => ({
+	useDispatchLobby: vi.fn(),
+}));
+
+//mock the LobbySelection component to better see what lobbies are passed down to the LobbySelection component
+vi.mock("../../components/LobbySelection/LobbySelection", () => ({
+	default: vi.fn(({ lobbies, joinLobby }) => {
+		return (
+			<div>
+				{lobbies.map((lobby: Lobby) => (
+					<Button
+						key={"test-button-" + lobby.id}
+						onClick={() => joinLobby(lobby)}
+					>
+						{lobby.id}:
+						{lobby.isLobbyFull() || lobby.isDisabled ? "disabled" : "enabled"}
+					</Button>
+				))}
+			</div>
+		);
+	}),
+}));
```

Comment:
We should maybe reconsider our testing approach for the pages if we require such complex mocks.

One approach would be, to move the "page integration tests" to playwright e2e tests.